docs/howto/operator/custom-operator.rst

```diff
@@ -0,0 +1,183 @@
+
+Let's extend our previous example to fetch name from MySQL:
+
+.. code:: python
+
+    class HelloDBOperator(BaseOperator):
+
+            @apply_defaults
+            def __init__(
+                    self,
+                    name: str,
+                    conn_id: str,
+                    database: str,
+                    *args, **kwargs) -> None:
+                super().__init__(*args, **kwargs)
+                self.name = name
+                self.conn_id = conn_id
+                self.database = database
+
+            def execute(self, context):
+                hook = MySqlHook(mysql_conn_id=self.mysql_conn_id,
+                         schema=self.database)
+                sql = "select name from user"
+                result = hook.get_first(sql)
+                message = "Hello {}".format(result['name'])
+                print(message)
+                return message
+
+When the operator invokes the query on the hook object, a new connection gets created if it doesn't exist. 
+The hook retrieves the auth parameters such as username and password from Airflow
+backend and passes the params to the :py:func:`airflow.hooks.base_hook.BaseHook.get_connection`. 
+
+
+User interface
+^^^^^^^^^^^^^^^
+Airflow also allows the developer to control how the operator shows up in the DAG UI.
+Override ``ui_color`` to change the background color of the operator in UI. 
+Override ``ui_fgcolor`` to change the color of the label.
+
+.. code::  python
+
+        class HelloOperator(BaseOperator):
+            ui_color = '#ff0000'
+            ui_fgcolor = '#000000'
+            ....
+
+Templating
+^^^^^^^^^^^
+You can use :ref:`Jinja templates <jinja-templating>` to parameterize your operator.
+Airflow considers the field names present in ``template_fields``  for templating while rendering
+the operator.
+
+.. code:: python
+    
+        class HelloOperator(BaseOperator):
+            
+            template_fields = ['name']
+            
+            @apply_defaults
+            def __init__(
+                    self,
+                    name: str,
+                    *args, **kwargs) -> None:
+                super().__init__(*args, **kwargs)
+                self.name = name
+
+            def execute(self, context):
+                message = "Hello from {}".format(name)
+                print(message)
+                return message
+
+        hello_task = HelloOperator(task_id='task_id_1', dag=dag, name='{{ task_id }}')
+
+In this example, Jinja looks for the ``name`` parameter and substitutes ``{{ task_id }}`` with
+task_id_1.
+
+The parameter can also contain a file name, for example, a bash script or a SQL file. You need to add
+the extension of your file in ``template_ext``. If a ``template_field`` contains a string ending with
+the extension mentioned in ``template_ext``, Jinja reads the content of the file and replace the templates
+with actual value.
+
+
+Define an operator extra link
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This information is already in the documentation. Do you want to repeat it?
https://airflow.readthedocs.io/en/latest/howto/define_extra_link.html

Reply:
So should I just guide the reader to that link or not mention this at all?

Reply:
Probably just link direct them to there.